src/main/scala/nl/knaw/dans/easy/pid/Command.scala

```diff
@@ -30,27 +31,38 @@ object Command extends App with DebugEnhancedLogging {
     verify()
   }
   val app = new PidGeneratorApp(new ApplicationWiring(configuration))
-  val result: Try[FeedBackMessage] = commandLine.subcommand match {
-    case Some(generate @ commandLine.generate) => generate.pidType() match {
-      case "doi" => app.generate(DOI)
-      case "urn" => app.generate(URN)
-    }
-    case Some(_ @ commandLine.runService) => runAsService()
-    case _ => Failure(new IllegalArgumentException(s"Unknown command: ${ commandLine.subcommand }"))
-  }
 
-  result.doIfSuccess(msg => println(s"OK: $msg"))
+  managed(app)
```

Comment:
To close the application properly we need to take error into account: even if an error occurs, we need to close the resource (database connection pool) properly. Therefore I made `PidGeneratorApp` extend the `Closeable` trait and wrapped it in a `resource.managed` here.